test/parallel_api/numeric/numeric.ops/exclusive_scan_by_segment.pass.cpp

```diff
@@ -200,22 +256,86 @@ DEFINE_TEST(test_exclusive_scan_by_segment)
     }
 };
 
-int main()
+template <typename _Tp>
+struct UserBinaryPredicate
+{
+    bool
+    operator()(const _Tp& __x, const _Tp& __y) const
+    {
+        using KeyT = ::std::decay_t<decltype(__y)>;
+        return __y != KeyT(1);
+    }
+};
+
+template <typename _Tp>
+struct UserBinaryOperation
+{
+    _Tp
+    operator()(const _Tp& __x, const _Tp& __y) const
+    {
+        return (__x < __y) ? __y : __x;
+    }
+};
+
+template <typename _Tp>
+struct UserBinaryOperation<::std::complex<_Tp>>
```

Comment:
Please move those predicates to some common place

Reply:
I have moved these to a new header called `utils_functors.h` in the support directory to store functors such as this.